halo2_backend/src/plonk/permutation/prover.rs

```diff
@@ -177,17 +176,13 @@ pub(in crate::plonk) fn permutation_commit<
             .commit_lagrange(&engine.msm_backend, &z, blind)
             .to_affine();
         let permutation_product_blind = blind;
-        let z = domain.lagrange_to_coeff(z);
-        let permutation_product_poly = z.clone();
-
-        let permutation_product_coset = domain.coeff_to_extended(z);
```

Comment:
Let me know if I understand this correctly.  This change has two improvements:
- 1. Previously the coeff to extend (IFFT) on the permutation product polynomial `z` was done one by one.  Now it's done in a batch.  This can bring a speed up if the batched IFFT is optimized.
- 2. Previously the coeff to extend (IFFT) on the permutation product polynomial `z` was calculated earlier and kept for a long time in memory.  Now it's done at the exact point where it's needed, and freed after it's used. This can bring a reduction in memory usage.

Reply:
yes, exactly.